docs/LANGUAGE.md

```diff
@@ -0,0 +1,388 @@
+TRUE   JSON true 
+FALSE  JSON false
+NULL   JSON null
+ALPHA  ASCII characters A-Z and a-z
+DIGIT  ASCII characters 0-9
+```
+
+## Opalog Queries
+
+OPA's policy engine stores data as structured documents. Conceptually there are two kinds of documents in OPA: **base documents** and **virtual documents**.
+
+**Base documents** contain raw data representing information from the application or environment which is relevant to policy.
+
+**Virtual documents** are defined by rules written by policy authors. Rules are written in terms of base documents and virtual documents. 
+
+Opalog does not distinguish between base documents and rule documents in queries. This allows policy authors to define virtual documents as abstractions on top of raw data and rules. Higher level rules are decoupled from underlying data sources which insulates them from schema changes.
+
+### Examples
+
+Many of the the examples in this section are based on the [Acme Corp](docs/data/acmecorp/README.md) data set. If you run OPA and load the Acme Corp data set, you can follow along. See the installation instructions in the data set for more details.
+
+### <a name="the-basics"></a> The Basics
+
+This section introduces the main elements of Opalog's syntax. The examples in this section do not rely on external data.
+
+The simplest type of rule is a constant value:
+
+```opalog
+pi :- 3.14159
+```
+
+We can query for the contents of the "pi" document produced by the rule above:
+
+```
+pi
+# 3.14159
+```
+
+Similarly, we can define rules which evaluate with only constants:
+
+```opalog
+v :- 42 = "the meaning of life"
+```
+
+If we query for the contents of "v" we see that the literal has been evaluated:
+
+```
+v
+# false
+```
+
+The order of operands in an equality literal does not matter:
+
+```opalog
+u :- "the meaning of life" = 42
+```
+
+The result is the same:
+
+```
+u
+# false
+```
+
+We can define rules in terms of variables as well:
+
+```opalog
+t :- x = 42, y = 41, x > y
+```
+
+Multiple literals are separated by the comma (",") character. In order for the rule to be true, all of the literals in the rule must true for some set of variable bindings. There may be multiple sets of variable bindings which make the rule true. The body of a rule can be understood intuitively as "\<literal-1> AND \<literal-2> AND ... AND \<literal-N>".
+
+When we query for the contents of "t" we see the obvious result:
+
+```
+t
+# true
+```
+
+In Opalog, the order of literals in a rule does not affect the content of the document:
+
+```opalog
+s :- x > y, y = 41, x = 42
+```
+
+The query result is the same:
+
+```
+s
+# true
+```
+
+Opalog supports references to nested documents. For example:
+
+```opalog
+r :- 
+  data = {"sites": [{"name": "prod"}, {"name": "smoke1"}, {"name": "dev"}]}, 
+  data.sites[].name = "prod"
+```
+
+The result:
+
+```
+r
+# true
+```
+
+We can generalize the above example by defining a **set rule** instead of a **boolean rule**:
+
+```opalog
+q[] = x :- 
```

Comment:
I worry about starting with a set example, just because it's the most confusing syntactically.  A dictionary or array would  be better.